### PR TITLE
feat: add action to button event

### PR DESCRIPTION
**Pull request by csiyang**
# Description

Adds associated button action and label to button click event. This is added so we can know on the backend what type of action the button was fired from.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/31822546/todos/5974836296)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

Preconditions
- Create a journey with buttons that have:
 > - NavigateAction
 > - NavigateToBlockAction
 > - NavigateToJourneyAction
 > - LinkAction with link as some chat platform (https://www.fb.me/zuck)
 > - LinkAction with link not to a chat platform (https://google.com)

Steps
- [ ] Visit the journey in the `journeys` project
- [ ] Open network terminal, as you click each of the buttons make sure to get the id of the event
- [ ] Search the id of the event in your [local arango](http://localhost:8529/_db/_system/_admin/aardvark/index.html#collection/events/documents/1)
- [ ] Events should have:
 > - NavigateAction: `action: NavigateAction`, `actionValue: undefined`
 > - NavigateToBlockAction: `action: NavigateToBlockAction`, `actionValue: getHeading of the step navigated to`
 > - NavigateToJourneyAction: `action: NavigateToJourneyAction`, `actionValue: the name of the journey`
 > - LinkAction (chat): `__typename: ChatOpenEvent`
 > - LinkAction (other): `action: LinkAction`, `actionValue: link of the button`

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

**Comment by csiyang** [comment]
Depends on https://github.com/JesusFilm/core/pull/1437

File: apps/api-journeys/src/app/modules/event/button/button.graphql [hunk 1]
```diff
@@ -36,6 +55,18 @@ type ButtonClickEvent implements Event {
...
   """
   value: String
+  """
+  Action type of the button when it was clicked
+  """
+  action: ButtonAction
+  """
+  The label for each corresponding action, mapping below:
+  NavigateAction - null
+  NavigateToBlockAction - StepName of the block
```
Comment (storyworks) [inline apps/api-journeys/src/app/modules/event/button/button.graphql:65]: StepId of the block? Same issue in line 33.

File: apps/api-journeys/src/app/modules/event/signUp/signUp.resolver.ts [hunk 2]
```diff
@@ -30,29 +30,38 @@ export class SignUpSubmissionEventResolver {
...
-      visitorId: visitor.id,
-      createdAt: new Date().toISOString(),
-      journeyId,
-      stepId: input.stepId,
-      label: null,
-      value: input.name,
-      email: input.email
-    })
+    const [SignUpSubmissionEvent] = await Promise.all([...promiseArray])
+    return SignUpSubmissionEvent
```
Comment (storyworks) [inline apps/api-journeys/src/app/modules/event/signUp/signUp.resolver.ts:65]: SignUp needs a test?

File: apps/api-journeys/src/app/modules/event/button/button.graphql (hunk 1 above)
Comment (storyworks) [reply apps/api-journeys/src/app/modules/event/button/button.graphql:65]: Or "Id of a StepBlock"?

File: libs/journeys/ui/src/components/Button/utils/getActionLabel/getActionLabel.spec.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  ButtonFields_action_NavigateToBlockAction as NavigateToBlockAction,
+  ButtonFields_action_NavigateToJourneyAction as NavigateToJourneyAction,
+  ButtonFields_action_LinkAction as LinkAction
+} from '../../__generated__/ButtonFields'
+import { TreeBlock } from '../../../../libs/block'
+import { TypographyVariant } from '../../../../../__generated__/globalTypes'
+import { getActionLabel } from './getActionLabel'
+
+describe('getActionLabel', () => {
+  it('should undefined for NavigateAction', () => {
```
Comment (storyworks) [inline libs/journeys/ui/src/components/Button/utils/getActionLabel/getActionLabel.spec.ts:12]: `should be undefined` ?

File: libs/journeys/ui/src/components/Button/Button.spec.tsx [hunk 4]
```diff
@@ -120,11 +124,25 @@ describe('Button', () => {
...
+      ...block,
+      action
+    }
+
     const result = jest.fn(() => ({
       data: {
         buttonClickEventCreate: {
           __typename: 'ButtonClickEvent',
-          id: 'uuiid'
+          id: 'uuiid',
```
Comment (storyworks) [inline libs/journeys/ui/src/components/Button/Button.spec.tsx:143]: I know it can be anything but did you mean `uuid`?

**Review by storyworks — COMMENTED** [review]
Just minor typos, otherwise all good. May as well fix if you have to merge latest main to this anyway.

I did notice no test for if journey slug is null. But from memory, NavigateToJourneyAction only gets set when journey is not null and you're only setting the extra `""` to fix linting... So should be ok without the test.

File: libs/journeys/ui/src/components/Button/utils/getActionLabel/getActionLabel.spec.ts [hunk 5]
```diff
@@ -99,6 +99,18 @@ describe('getActionLabel', () => {
     expect(result).toEqual(action.journey?.slug)
   })
 
+  it('should return Unknwon Journey for NavigateToJourneyAction if journey is null', () => {
```
Comment (storyworks) [inline libs/journeys/ui/src/components/Button/utils/getActionLabel/getActionLabel.spec.ts:102]: When you merged latest main fix `Unknown`?

**Comment by csiyang** [comment]
Merged into https://github.com/JesusFilm/core/pull/1453

**Pull request closed without merging**